File: src/main/java/org/hwabeag/divesystem/expansions/DiveSystemExpansion.java
package org.hwabeag.divesystem.expansions;

import me.clip.placeholderapi.expansion.PlaceholderExpansion;
import org.bukkit.OfflinePlayer;
import org.bukkit.configuration.file.FileConfiguration;
import org.hwabeag.divesystem.DiveSystem;
import org.hwabeag.divesystem.config.ConfigManager;
import org.jetbrains.annotations.NotNull;

import java.util.Objects;

public class DiveSystemExpansion extends PlaceholderExpansion {


    FileConfiguration PlayerConfig = ConfigManager.getConfig("player");
    private final DiveSystem plugin; //

    public DiveSystemExpansion(DiveSystem plugin) {
        this.plugin = plugin;
    }

    @Override
    public boolean persist() {
        return true;
    }

    @Override
    public boolean canRegister() {
        return true;
    }

    @Override
    public @NotNull String getAuthor() {
        return plugin.getDescription().getAuthors().toString();
    }

    @Override
    public @NotNull String getIdentifier() {
        return "dive_system";
    }

    @Override
    public @NotNull String getVersion() {
        return plugin.getDescription().getVersion();
    }

    @Override
    public String onRequest(OfflinePlayer player, @NotNull String params) {
        if (params.equalsIgnoreCase("get-index")) { // %dive_system_get-index%
            String name = Objects.requireNonNull(player).getName();
            if (PlayerConfig.getString(name + ".잠수") != null) {
                return PlayerConfig.getString(name + ".잠수");
            }
        }
        return null; //
    }
}

File: src/main/kotlin/org/hwabeag/divesystem/expansions/DiveSystemExpansion.kt
package org.hwabeag.divesystem.expansions

import me.clip.placeholderapi.expansion.PlaceholderExpansion
import org.bukkit.OfflinePlayer
import org.hwabeag.divesystem.DiveSystem
import org.hwabeag.divesystem.data.DatabaseManager

class DiveSystemExpansion(private val plugin: DiveSystem) : PlaceholderExpansion() {
    override fun persist(): Boolean = true

    override fun canRegister(): Boolean = true

    override fun getAuthor(): String = plugin.description.authors.toString()

    override fun getIdentifier(): String = "dive_system"

    override fun getVersion(): String = plugin.description.version

    override fun onRequest(player: OfflinePlayer?, params: String): String? {
        if (!params.equals("get-index", ignoreCase = true)) {
            return null
        }
        val name = player?.name ?: return null
        return DatabaseManager.getPoints(name).toString()
    }
}
